Small cleanup on BMPImageReader.cpp

Ahmad-S792 opened the pull request:
#### d156132fce2fe3a0d90b92c2866ca4776c473316
<pre>
Small cleanup on BMPImageReader.cpp

<a href="https://bugs.webkit.org/show_bug.cgi?id=253123">https://bugs.webkit.org/show_bug.cgi?id=253123</a>
rdar://problem/106387835

Reviewed by Fujii Hironori.

Merge: <a href="https://chromium.googlesource.com/chromium/blink/+/a057769943417c61acd574458034ac6546ed6f6a">https://chromium.googlesource.com/chromium/blink/+/a057769943417c61acd574458034ac6546ed6f6a</a>

&apos;m_headerOffset + m_infoHeader.biSize&apos; is used often in BMPImageReader.cpp.
It is even recalulated several time in a same function. This patch cleans up
this kind of repeated code.

* Source/WebCore/platform/image-decoders/bmp/BMPImageReader.cpp:
(BMPImageReader::BMPImageReader):
(BMPImageReader::decodeBMP):
(BMPImageReader::readInfoHeaderSize):
(BMPImageReader::processBitmasks):
(BMPImageReader::processColorTable):
* Source/WebCore/platform/image-decoders/bmp/BMPImageReader.h: Remove &apos;m_tableSizeInBytes&apos;

Canonical link: <a href="https://commits.webkit.org/265704@main">https://commits.webkit.org/265704@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/58881a704a53c87d7bb9c87cfa74488cf429eec9

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/11642 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/11842 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/12210 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/13284 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/11087 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/11661 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/14230 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/11822 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/13963 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/11806 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/12654 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/9878 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/13707 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/9946 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/10574 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/17706 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/11020 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/10728 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/13903 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/11119 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/9171 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/10296 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/2802 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/14576 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/10976 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aprotyas commented on Source/WebCore/platform/image-decoders/bmp/BMPImageReader.cpp:
```diff
@@ -392,20 +394,23 @@ bool BMPImageReader::processBitmasks()
+        const size_t headerEnd = m_headerOffset + m_infoHeader.biSize;
+        const size_t bitmasksSize = 12;
+        const size_t bitmasksEnd = headerEnd + bitmasksSize;
         static const size_t SIZEOF_BITMASKS = 12;
```
This definition can be removed now

Ahmad-S792 replied:
Forgot.. Good catch!

Ahmad-S792 replied:
Fixed in [a867c98](https://github.com/WebKit/WebKit/commit/a867c98a3ae1b3a76497fcd68dab5391896454dc)

aprotyas commented on Source/WebCore/platform/image-decoders/bmp/BMPImageReader.cpp:
```diff
@@ -392,20 +394,23 @@ bool BMPImageReader::processBitmasks()
 
         // Fail if we don't have enough file space for the bitmasks.
+        const size_t headerEnd = m_headerOffset + m_infoHeader.biSize;
+        const size_t bitmasksSize = 12;
```
Consider making this a `static constexpr size_t`

Ahmad-S792 replied:
Will do.

Ahmad-S792 replied:
Fixed in [a867c98](https://github.com/WebKit/WebKit/commit/a867c98a3ae1b3a76497fcd68dab5391896454dc).

aprotyas commented on Source/WebCore/platform/image-decoders/bmp/BMPImageReader.cpp:
```diff
@@ -462,10 +467,11 @@ bool BMPImageReader::processBitmasks()
 
 bool BMPImageReader::processColorTable()
 {
-    m_tableSizeInBytes = m_infoHeader.biClrUsed * (m_isOS21x ? 3 : 4);
```
We can't remove this line, we're setting a data member here

Ahmad-S792 replied:
But I moved it to line 470 as:
`const size_t tableSizeInBytes = m_infoHeader.biClrUsed * (m_isOS21x ? 3 : 4);`

aprotyas replied:
Yes, that part is correct, but the class `BMPImageReader` has a property `size_t m_tableSizeInBytes`. If we remove this line, we will not be updating that property anymore. At that point this changes the semantics of the code.

In the future if someone adds a function that reads `m_tableSizeInBytes` expecting it to have been updated by `BMPImageReader::processColorTable()`, they will have problems.

Alternatively you can remove the `m_tableSizeInBytes` data member from the `BMPImageReader` class and keep the changes you have as-is, though someone more familiar with this class should comment on that.

Ahmad-S792 replied:
Ops! I got it. Pushed new [f6a20e4](https://github.com/WebKit/WebKit/commit/f6a20e490aaaacb07ef5f6d5bad425110e70e629).

Ahmad-S792 committed:
Small cleanup on BMPImageReader.cpp

https://bugs.webkit.org/show_bug.cgi?id=253123
rdar://problem/106387835

Reviewed by Fujii Hironori.

Merge: https://chromium.googlesource.com/chromium/blink/+/a057769943417c61acd574458034ac6546ed6f6a

'm_headerOffset + m_infoHeader.biSize' is used often in BMPImageReader.cpp.
It is even recalulated several time in a same function. This patch cleans up
this kind of repeated code.

* Source/WebCore/platform/image-decoders/bmp/BMPImageReader.cpp:
(BMPImageReader::BMPImageReader):
(BMPImageReader::decodeBMP):
(BMPImageReader::readInfoHeaderSize):
(BMPImageReader::processBitmasks):
(BMPImageReader::processColorTable):
* Source/WebCore/platform/image-decoders/bmp/BMPImageReader.h: Remove 'm_tableSizeInBytes'

Canonical link: https://commits.webkit.org/265704@main
